plugins/out_stackdriver/stackdriver.c

```diff
@@ -330,6 +330,72 @@ static int cb_stackdriver_init(struct flb_output_instance *ins,
     return 0;
 }
 
+static int validate_severity_level(const char * severity)
+{
+    int i = 0;
+
+    const char * severity_level[] = {
+        "EMERGENCY",
+        "ALERT",
+        "CRITICAL",
+        "ERROR",
+        "WARNING",
+        "NOTICE",
+        "INFO",
+        "DEBUG",
+        "DEFAULT",
+        NULL
+    };
+
+    if (severity == NULL) {
+        return -1;
+    }
+
+    for (i = 0; severity_level[i] != NULL; i++) {
+        if (strcasecmp(severity, severity_level[i]) == 0) {
+            return 0;
+        }
+    }
+    return -1;
+}
+
+static int map_get_str_kv(flb_sds_t buf, const msgpack_object *o,
+                          const flb_sds_t key)
+{
+    int i = 0;
+    msgpack_object_kv * p = NULL;
+
+    if (o == NULL || o->type != MSGPACK_OBJECT_MAP || buf == NULL) {
+        return -1;
+    }
+
+    for (i = 0; i < o->via.map.size; i++) {
+        p = &o->via.map.ptr[i];
+        if (p->key.type != MSGPACK_OBJECT_STR) {
+            continue;
+        }
+
+        if (p->val.via.str.size > 10) {
+            continue;
+        }
+
+        if (flb_sds_cmp(key, p->key.via.str.ptr, flb_sds_len(key)) == 0) {
```

Comment:
if key == "testAAAA" and  p->key.via.str.ptr == "testAAAABBBB" it will match, instead of pass flb_sds_len() pass p->key.via.str.size